api/src/utils/redact-object.ts

```diff
@@ -111,7 +111,18 @@ export function getReplacer(replacement: Replacement, values?: Values) {
 		? Object.entries(values).filter(([_k, v]) => typeof v === 'string' && v.length > 0)
 		: [];
 
+	const seen = new WeakSet();
+
 	return (_key: string, value: unknown) => {
+		// Skip circular values
+		if (typeof value === 'object' && value !== null) {
```

Comment:
Thought of catching "truthy" values instead of `null` only.
```suggestion
		if (typeof value === 'object' && value) {
```

Reply:
Random tangent that popped up in my mind, what if we use `isObject` util function which I assume was the original intention (aside from the extra Array check) of this IF statement?

https://github.com/directus/directus/blob/e78a01013a15af22c37c28b4a040ce12d2b4df30/api/src/utils/redact-object.ts#L99

especially since it's used in this same redactObject function already at line 99, which means doing `if (isObject(value))` instead

Reply:
We only need to track objects as other values cannot be circular, the null check ensures that by filtering out `Array` and `null` (which are also typeof object).

Reply:
> Random tangent that popped up in my mind, what if we use `isObject` util function which I assume was the original intention (aside from the extra Array check) of this IF statement?
> 
> https://github.com/directus/directus/blob/e78a01013a15af22c37c28b4a040ce12d2b4df30/api/src/utils/redact-object.ts#L99
> 
> especially since it's used in this same redactObject function already at line 99, which means doing `if (isObject(value))` instead

Good idea ❤️